### PR TITLE
Updated Hilla React tutorial

### PR DESCRIPTION
**Pull request by marcushellberg**
@westonhancock - here's a PR with the updated tutorial. The URL is the same as the old one, so that should be fine. We want to retire https://hilla.dev/docs/lit/start/in-depth and all sub-pages and redirect them to this one.

### CODE REVIEW

File: articles/react/start/basics/index.adoc [hunk 1]
```diff
@@ -1,8 +1,391 @@
...
+-- `routing.tsx` is routing setup for the application.
+
+Second, there's `src`, which is the Spring Boot backend:
+- `com.example.application`
+-- `Application.java` is the main Spring Boot application class;
+-- `data` contains all the database-access-related code;
+--- `Contact.java` and `Company.java` are the JPA entities;
+--- `ContactRepository.java` and `CompanyRepository.java` are repositories for accessing the database; and
+--- `DataInitializer.java` populates the database with example data.
+-- `services` is where you'll create a backend service for the application.
```
Comment (tarekoraby) [inline articles/react/start/basics/index.adoc:66]: @russelljtdyer, the changes you introduced make the bullets in the paragraph look proper. I much prefer Marcus's original formulation (except that there is a mistake in the original list, as it should `routes.tsx`, NOT "routing").

```suggestion
* `frontend` - contains the React frontend
** `views`
*** `MainLayout.tsx` is the parent layout that includes the navigation bar
*** `contacts`
**** `ContactsView.tsx` is the view we are implementing
** `routes.tsx` is routing setup for the app
* `src` is the Spring Boot backend
** `com.example.application`
*** `Application.java` is the main Spring Boot application class
*** `data` contains all the database-access-related code
**** `Contact.java` and `Company.java` are the JPA entities
**** `ContactRepository.java` and `CompanyRepository.java` are repositories for accessing the database
**** `DataInitializer.java` populates the database with example data 
*** `services` is where we will create a backend service for the application
```

@marcushellberg , company and companyRepo are present in the starter, but not mentioned in the list.

File: articles/react/start/basics/index.adoc [hunk 2]
```diff
@@ -1,8 +1,391 @@
...
+== Run the Hilla Application
+
+There are two ways to run the application: you can use your IDE to run `Application.java` -- this is recommended since it supports debugging; or you can run the default Maven goal from the command-line (i.e., `mvn`).
+
+Hilla will automatically reload changes to frontend and backend code when saving. IntelliJ requires you manually to build the project after making changes to Java code, while VS Code will  build automatically the project on save.
+
+
+== Create a Browser-Callable Java Service
+
+Hilla makes it possible to call Java classes from TypeScript with the `@BrowserCallable` annotation. To do this, create a `CRMService.java` class in the `com.example.application.service` package. The service handles both listing and saving Contacts. Add the following code to the class:
```
Comment (tarekoraby) [inline articles/react/start/basics/index.adoc:78]:
```suggestion
Hilla makes it possible to call Java classes from TypeScript with the `@BrowserCallable` annotation. To do this, create a `CRMService.java` class in the `com.example.application.services` package. The service handles both listing and saving Contacts. Add the following code to the class:
```

File: articles/react/start/basics/index.adoc [hunk 3]
```diff
@@ -1,8 +1,391 @@
...
+- The service injects `ContactRepository` and `CompanyRepository` in the constructor for database access.
+- This defines DTOs for the view as Java Records, including validation annotations that you want to enforce, both in the UI and the service.
+- The service defines the CRUD methods needed for the CRM. 
+
+Now, you'll have to build the application. Hilla will generate the needed TypeScript for accessing the service.
+
+
+== Listing Contacts in a Data Grid
+
+With the backend completed, you can start building the UI. Change the contents of `ContactsView.tsx` to the following:
```
Comment (tarekoraby) [inline articles/react/start/basics/index.adoc:192]:
```suggestion
With the backend completed, you can start building the UI. Change the contents of `Frontend/views/contacts/ContactsView.tsx` to the following:
```

File: articles/react/start/basics/index.adoc (hunk 1 above)
Comment (marcushellberg) [reply articles/react/start/basics/index.adoc:66]: They are there on the same line as contact and contact repo

**Review by tarekoraby — APPROVED** [review]

**Pull request merged**